src/expression/evaluation_context.ts

```diff
@@ -20,7 +20,9 @@ export class EvaluationContext {
         this.feature = null;
         this.featureState = null;
         this.formattedSection = null;
-        this._parseColorCache = {};
+        // the cache keys are user controlled (from the source JSON), so
+        // avoid prototype pollution by creating a record with a null prototype
+        this._parseColorCache = Object.create(null) as {[_: string]: Color};
```

Comment:
The current TypeScript configs didn't seem to care whether this `any` was casted or not, but I added it so that it doesn't need to be fixed in the future if the TS configs are ever made stricter in the future.

Reply:
If you think a Map object would be a better fit here (looks like it, since we are talking about cache) please change the code accordingly.

Reply:
Sure, I think it's the best choice here, but was hesitant to introduce it as the first use in this repo. I thought it might be a ES5 compatibility thing, but on further investigation the codebase is already using ES6 features like `Set` and `Number.isNaN` so it's probably just a matter of this being a long-lived codebase.

Changed to use a `Map` in c52614560a3954ab939200ee98d6e309451b902d.